api/bindings.h

```diff
@@ -61,6 +61,22 @@ typedef struct cache_t {
 
 } cache_t;
 
+/**
+ * A view into an externally owned byte slice (Go `[]byte`).
+ * Use this for the current call only. A view cannot be copied for safety reasons.
+ * If you need a copy, use [`to_owned`].
+ *
+ * Go's nil value is fully supported, such that we can differentiate between nil and an empty slice.
+ */
+typedef struct ByteSliceView {
+  /**
+   * True if and only if the byte slice is nil in Go. If this is true, the other fields must be ignored.
+   */
+  bool is_nil;
```

Comment:
Not sure if it's a good idea, but what about using `Option<ByteSliceView>` for this?

Reply:
Since we don't know how the memory layout of Rust's `Option<ByteSliceView>` is, we cannot easily get a counter-part in Go.

The only thing we have is

> If T is an FFI-safe non-nullable pointer type, Option<T> is guaranteed to have the same layout and ABI as T and is therefore also FFI-safe. As of this writing, this covers &, &mut, and function pointers, all of which can never be null.

(https://doc.rust-lang.org/nomicon/other-reprs.html), which is a way to represent nullable pointers.